code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx

```diff
@@ -45,7 +45,7 @@ import type { StorySpecifier } from '../store/StoryIndexStore';
 const globalWindow = globalThis;
 
 function focusInInput(event: Event) {
-  const target = event.target as Element;
+  const target = event.composedPath()[0] as Element;
```

Comment:
Hey @stropitek thank you so much for your contribution! I don't know if this would work in 100% of our use cases, so I wonder if we need to make this change safer?

I've seen this being handled other codebases like so:

```ts
const target = (event.composedPath && event.composedPath()[0]) || event.target
```

```ts
const target = (evnt.target.shadowRoot && evnt.composed) ? (evnt.composedPath()[0] || evnt.target) : evnt.target
```

```ts
const target = event.composedPath ? event.composedPath()[0] || event.target : event.target
```

Reply:
I made the assumption from https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath that there was enough browser support for this to be safe. It's also supposed to work both in closed and open modes, see https://mdn.github.io/web-components-examples/composed-composed-path/

But I agree that it does not "feel" safe because of the function call and the property access.

I'm happy to change it + add a test case for the `event.target` case.


> ```ts
> const target = (event.composedPath && event.composedPath()[0]) || event.target
> ```
> 

I'd go with this one since it is the least tied to any browser spec.

Reply:
I think support is well within the range of what storybook supports, but considering it's a 1-liner to add the fallback I think we should go with the above suggestion @stropitek